library/general/gatewayclassnamespaces/constraint.yaml

```diff
@@ -0,0 +1,16 @@
+apiVersion: constraints.gatekeeper.sh/v1beta1
+kind: GatewayClassNamespaces
+metadata:
+  name: gatewayclassnamespaces
+spec:
+  match:
+    kinds:
+      - apiGroups: ["networking.x-k8s.io"]
+        kinds: ["Gateway"]
+  parameters:
+    gatewayClassNamespaces:
+      external-lb:
```

Comment:
The k8s API guidelines discourages use of arbitrarily-keyed maps:

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps

Reply:
Good call, turned this into a list.